integration/tests/cook/test_basic.py

```diff
@@ -213,19 +215,34 @@ def test_max_runtime_exceeded(self):
         job_timeout_interval_ms = job_timeout_interval_seconds * 1000
         max_runtime_ms = 5000
         assert max_runtime_ms < job_timeout_interval_ms
-        job_uuid, resp = util.submit_job(self.cook_url, command='sleep %s' % job_timeout_interval_seconds,
-                                         executor=job_executor_type, max_runtime=max_runtime_ms)
-        self.assertEqual(201, resp.status_code, msg=resp.content)
-        util.wait_for_job(self.cook_url, job_uuid, 'completed', job_timeout_interval_ms)
-        job = util.load_job(self.cook_url, job_uuid)
-        self.assertEqual(1, len(job['instances']))
-        self.assertEqual('failed', job['instances'][0]['status'])
-        self.assertEqual(2003, job['instances'][0]['reason_code'])
-        self.assertEqual('Task max runtime exceeded', job['instances'][0]['reason_string'])
-        if job_executor_type == 'cook':
-            job = util.wait_for_exit_code(self.cook_url, job_uuid)
-            self.assertEqual(-15, job['instances'][0]['exit_code'])
-            self.assertTrue(bool(job['instances'][0]['sandbox_directory']))
+        try:
+            job_uuid, resp = util.submit_job(self.cook_url,
+                                             command=f'sleep {job_timeout_interval_seconds}',
+                                             executor=job_executor_type, max_runtime=max_runtime_ms)
+            self.assertEqual(201, resp.status_code, msg=resp.content)
+            util.wait_for_job(self.cook_url, job_uuid, 'running')
+            util.wait_for_job(self.cook_url, job_uuid, 'completed', job_timeout_interval_ms)
+            job = util.wait_for_end_time(self.cook_url, job_uuid)
+            job_details = f"Job details: {json.dumps(job, sort_keys=True)}"
+            self.assertEqual(1, len(job['instances']), job_details)
+            instance = job['instances'][0]
+            self.assertEqual('failed', instance['status'], job_details)
+            valid_reasons = [# we killed the job because it took too long
+                             CookTest.REASON_MAX_RUNTIME_EXCEEDED,
+                             # the job was killed before the executor was properly registered
+                             CookTest.REASON_EXECUTOR_UNREGISTERED,
+                             # the job didn't die gracefully when we terminated it
```

Comment:
Why would this happen?

Reply:
This is one of the exit codes reported in #299. I'm actually thinking now that we shouldn't accept either of these other reasons until we figure out why they happen. Since the assertions now print the full job details string, we should have some extra info for debugging this if they come up again. However, if these were happening because of a scheduling delay issue (which is the only reason that I can think of where we'd be getting the `REASON_EXECUTOR_UNREGISTERED` error), then the more robust wait call chain we're using now may resolve this.

Reply:
:+1: